motr/cob.c

```diff
@@ -1848,18 +1849,49 @@ M0_INTERNAL int m0__obj_namei_send(struct m0_op_obj *oo)
 	}
 	cr->cr_cob_attr = cob_attr;
 
-	/* Set layout id for CREATE op.*/
+	/* Skip cob lookup if obj.ob_attr.oa_pver is not empty.
+	 * We are assuming here, if pver is not empty that means we
+	 * have received pver from s3 */
+	skip_lookup = false;
+	obj = m0__obj_entity(oo->oo_oc.oc_op.op_entity);
+	if ((cr->cr_opcode == M0_EO_GETATTR) &&
+	     m0_fid_is_set(&obj->ob_attr.oa_pver) &&
+	     m0_fid_is_valid(&obj->ob_attr.oa_pver)) {
+		skip_lookup = true;
+	}
+
+	/* Set layout id and pver for CREATE op.*/
 	if (cr->cr_opcode == M0_EO_CREATE) {
-		obj = m0__obj_entity(oo->oo_oc.oc_op.op_entity);
 		cr->cr_cob_attr->ca_lid = obj->ob_attr.oa_layout_id;
+		 if (obj->ob_entity.en_app_type != M0_OOF_MOTR_APP) {
+			/* For create operation setting up pool version locally
+			* found in pools common, so cob lookup call to server
+			* can be skipped */
+			obj->ob_attr.oa_pver.f_container = pv->pv_id.f_container;
+			obj->ob_attr.oa_pver.f_key = pv->pv_id.f_key;
+			skip_lookup = true;
```

Comment:
By skipping cob_create fops during entity create, CROW(Create On Write) would be used to send cob_create fops during write op, has this case been tested?

Reply:
This lookup skip is only for meta-data not for data.

Reply:
By skipping during M0_EO_CREATE, we are skipping sending cob_create fops which creates the cobs on server, not just store pver and lid. Now those cobs will be created during M0_OC_WRITE operation and where the cob create fops will be sent first and then iofops for write will be sent.

Please test if not already done so, the case when cob fops are skipped during write as added by your patch.

Reply:
Cob create fops will be send only for the units which are not spanned the parity group.
And both iofop and cob create fops will be sent parallely.

Reply:
Thanks for the correction Madhav.